mat64/symmetric.go

```diff
@@ -237,7 +236,30 @@ func (s *SymDense) SymRankK(a Symmetric, alpha float64, x Matrix) {
 //  s = x * x'
 func (s *SymDense) SymOuterK(x Matrix) {
 	r, _ := x.Dims()
-	s.reuseAs(r)
+
```

Comment:
The problem here is that there is no test for s == x. If this is the case, then zeroing s also zeros x